src/index.ts

```diff
@@ -834,9 +842,13 @@ export function collect(...paths: string[]): PropertyDecorator {
  */
 export function diff(...paths: string[]): PropertyDecorator {
   return macro(obj => {
-    let arrays = getPaths(obj, paths);
+    if (paths.length === 0) {
+      return [];
+    }
+
+    let arrays = getArrays(obj, paths);
 
-    let intersect = arrays.shift();
+    let intersect = arrays.shift() as any[];
```

Comment:
TS hinted me now that the return value here could be undefined, so I added the length check before to make sure it can't be undefined.